src/main/java/org/spongepowered/api/data/manipulator/mutable/common/AbstractBooleanData.java

```diff
@@ -42,17 +42,27 @@
 public abstract class AbstractBooleanData<M extends DataManipulator<M, I>, I extends ImmutableDataManipulator<I, M>> extends
         AbstractSingleData<Boolean, M, I> {
 
-    private final boolean defaultValue;
-
+    /**
+     * @deprecated Use {@link #AbstractBooleanData(Key, boolean, boolean)} instead.
+     */
+    @Deprecated
+    @SuppressWarnings("unchecked")
     protected AbstractBooleanData(boolean value, Key<? extends BaseValue<Boolean>> usedKey, boolean defaultValue) {
-        super(value, usedKey);
-        this.defaultValue = defaultValue;
+        this((Key<Value<Boolean>>) usedKey, value, defaultValue);
+    }
+
+    protected AbstractBooleanData(Key<Value<Boolean>> usedKey, boolean value) {
+        super(usedKey, value);
+    }
+
+    protected AbstractBooleanData(Key<Value<Boolean>> usedKey, boolean value, boolean defaultValue) {
+        super(usedKey, value, defaultValue);
     }
 
     @SuppressWarnings("unchecked")
     @Override
     protected Value<Boolean> getValueGetter() {
-        return Sponge.getRegistry().getValueFactory().createValue((Key<Value<Boolean>>) this.usedKey, this.getValue(), this.defaultValue);
+        return Sponge.getRegistry().getValueFactory().createValue((Key<Value<Boolean>>) this.usedKey, getValue(), this.defaultValue);
```

Comment:
Maybe use `this.value` instead of `getValue()` here (and also on similar occurancaes)? Similar to `AbstractSingleCatalogData`.

Reply:
Or as an alternative move the implementation to `AbstractSingleData` entirely?